daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -362,13 +368,13 @@ object Node {
             signatories2,
             stakeholders2,
             key2,
-            _,
+            byKey2,
             ) =>
           import nf._
           coid === coid2 && templateId == templateId2 &&
           actingParties.forall(_ => actingParties == actingParties2) &&
           signatories == signatories2 && stakeholders == stakeholders2 &&
-          key.forall(_ => key == key2)
+          key.forall(_ => key == key2) && byKey.forall(_ => byKey == byKey2)
```

Comment:
Please make sure that you also cover these cases in your other PR #7625 .

Reply:
:+1: